src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let stateClone = { ...state };
+  const list = [];
+
+  function removeProperties(obj, keysToRemove) {
+    for (let index = 0; index < keysToRemove.length; index++) {
+      delete stateClone[keysToRemove[index]];
+      delete obj[keysToRemove[index]];
+    }
+
+    return obj;
+  }
 
+  for (let index = 0; index < actions.length; index++) {
+    const item = actions[index];
+
+    switch (item.type) {
+      case 'addProperties':
+        stateClone = { ...stateClone, ...item.extraData };
+        list.push({ ...stateClone, ...item.extraData });
```

Comment:
There's no need to spread 'item.extraData' again when pushing to the list since 'stateClone' already includes 'item.extraData' from the previous spread operation on line 27.